src/linker/Linker/TypeReferenceExtensions.cs

```diff
@@ -8,6 +7,53 @@ namespace Mono.Linker
 {
 	public static class TypeReferenceExtensions
 	{
+		public static string GetDisplayName (this TypeReference type)
+		{
+			if (type == null)
+				return string.Empty;
+
+			var builder = new System.Text.StringBuilder ();
+			if (type.DeclaringType != null)
+				builder.Append (type.DeclaringType.GetDisplayName ()).Append (".");
+
+			if (type.HasGenericParameters) {
+				builder.Append (type.Name.Substring (0, type.Name.IndexOf ('`')));
+				builder.Append ('<');
+				for (int i = 0; i < type.GenericParameters.Count - 1; i++)
+					builder.Append ($"{type.GenericParameters[i]},");
```

Comment:
Better pattern is not to string.concant before calling into StringBuilder (you have that at many places)